#166789956 Users Should Create And Update an Article

kodek-sleuth opened the pull request:
#### What does this PR do?
Should enable a user to create and update an article

#### Description of Task to be completed
- Setup redux to manage state changes
- Setup components and dispatch functions

#### How should this be manually tested?
1. Run the following commands in the terminal
```
Clone the project
 cd ah-92explorers-frontend
 git checkout ft-create-update-articles-166789956
`npm i` to install all dependencies 
`npm run dev` and navigate to localhost:8080/article and localhost:8080/article/:articleSlug/update
```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166789956](https://www.pivotaltracker.com/story/show/166789956)
#### Screenshots (if appropriate)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

Commit:
feat(user-login): users should be able to Login

- add a login route
- add login component
- add login container
- validate inputs
- consume login endpoint from the backend
- manage states
- redirect routes
- add tests

[Start #166789953]

Commit:
feat(user-signup): enable user to sign up
 - enable user signup
 - send account t verification email
 - validate user information
 - write test

[Finishes #166789951]

Commit:
feat(create-update-articles): users should create and update an article

- create update and insertion components
- implement redux
- test components

[Finishes #166789956]

Commit:
feat(create-update-articles): users should create and update an article

- create update and insertion components
- implement redux
- test components

[Finishes #166789956]

Hervera reviewed (changes requested):
Fix the CircleCI build error

elemanhillary-zz reviewed (changes requested):
kindly fix hound issues in your test files

elemanhillary-zz commented on .env.local:
```diff
@@ -0,0 +1,3 @@
+REACT_APP_CLOUDINARY_NAME=authors-haven-92-explorers
+REACT_APP_CLOUDINARY_KEY=732819359484416
+REACT_APP_CLOUDINARY_SECRET=NoV0BGAyaX5ATd-e74wIPtGmwaY
```
kindly insert a new line at `EOF`

elemanhillary-zz commented on src/__tests__/components/CreateArticle.test.js:
```diff
@@ -0,0 +1,45 @@
+    expect(wrapper.find('.mainDiv').length).toBe(1);
+  });
+  it('should render a publishBtn class', () => {
+    expect(wrapper.find('.publishBtn').length).toBe(1);
```
as a `TTL` forexample i need to show me that since you are testing components and using `find` to find button elements, i want you to go a head and simulate the `click` event to test if your buttons respond to `events`.

elemanhillary-zz commented on src/helpers/ckEditorConfig.js:
```diff
@@ -0,0 +1,46 @@
+import axios from 'axios';
+
+// Function That will return a promise object when resolved
+export const getImageUrl = (file) => new Promise((resolve, reject) => {
```
kindly change this `getImageUrl` to use `es6` pattern of `async/await` instead of `new Promise`

elemanhillary-zz commented on src/redux/actions/actionCreators/articles.js:
```diff
@@ -0,0 +1,46 @@
+export const publishArticle = (articleData) => async (dispatch) => {
+  try {
+    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjIwYjlhZGQ4LTM1ZTUtNDgwNy04M2FlLWZlY2Q1YjAxMjNhOSIsImVtYWlsIjoiY2VsZXN0aW40M0BnbWFpbC5jb20iLCJ1c2VybmFtZSI6Im1hcmxleSIsImlhdCI6MTU2NjU3MTcwNywiZXhwIjoxNTY2NjU4MTA3fQ.EK6hSflXBLL6TqoxmPdRhmrG4-tlPl2Kaa7McQ0BAAc';
+    const article = await axios.post('http://127.0.0.1:3000/api/articles', articleData, { headers: { Authorization: `Bearer ${token}` } });
```
kindly donot hardcode `urls` in your project it would be better if you make use of the `.env` file

elemanhillary-zz commented on src/redux/actions/actionCreators/articles.js:
```diff
@@ -0,0 +1,46 @@
+export const updateArticle = (articleData) => async (dispatch) => {
+  try {
+    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjIwYjlhZGQ4LTM1ZTUtNDgwNy04M2FlLWZlY2Q1YjAxMjNhOSIsImVtYWlsIjoiY2VsZXN0aW40M0BnbWFpbC5jb20iLCJ1c2VybmFtZSI6Im1hcmxleSIsImlhdCI6MTU2NjU3MTcwNywiZXhwIjoxNTY2NjU4MTA3fQ.EK6hSflXBLL6TqoxmPdRhmrG4-tlPl2Kaa7McQ0BAAc';
+    const article = await axios.put('http://127.0.0.1:3000/api/articles/lessh1greaterwrite-a-titlelessh1greater-47', articleData, { headers: { Authorization: `Bearer ${token}` } });
```
the same here.

elemanhillary-zz commented on src/redux/actions/actionTypes/articles.js:
```diff
@@ -0,0 +1,5 @@
+export const CREATE_ARTICLE = 'CREATE_ARTICLE';
```
consider refactoring this to, atleast it makes it abit clean but its an option.
```javascript
export const articleTypes = {
    CREATE_ARTICLE = 'CREATE_ARTICLE';
    UPDATE_ARTICLE = 'UPDATE_ARTICLE';
    GET_ARTICLE = 'GET_ARTICLE';
    FAILED_ARTICLE_CREATION = 'FAILED_ARTICLE_CREATION';
    FAILED_ARTICLE_UPDATE = 'FAILED_ARTICLE_UPDATE';
}
```

niyoceles reviewed (changes requested):
Kindly add the screenshot could be helpful

elemanhillary-zz commented on src/__tests__/components/CreateArticle.test.js:
```diff
@@ -0,0 +1,48 @@
+  it('should publish article on click', () => {
+    const btn = wrapper.find('.publishBtn');
+    const component = wrapper.instance();
+    btn.simulate('click')
```
what happens after the simulated click event

elemanhillary committed:
feat(user-login): users should be able to login with social media accounts

- add a login route
- add login component
- add login container
- validate inputs
- consume login endpoint from the backend
- manage states
- redirect routes
- add tests

[Delivers #166789953]

Hervera reviewed (changes requested):
You have too many commits
